src/Raven.Server/Documents/Handlers/DocumentHandler.cs

```diff
@@ -222,7 +223,9 @@ private async Task GetDocumentsByIdAsync(DocumentsOperationContext context, Micr
                 }
                 if (document == null && ids.Count == 1)
                 {
-                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    HttpContext.Response.StatusCode = GetStringFromHeaders("If-None-Match") == HttpCache.NotFoundResponse
```

Comment:
When requesting multiple documents that do not exist we respond with `NotModified` (if the previous response was the same) but when requesting only one we respond with `NotFound`.